source-boilerplate/boilerplate.go

```diff
@@ -29,6 +29,11 @@ import (
 	"google.golang.org/grpc/metadata"
 )
 
+// StateKey is used to key binding-specific state a connector's driver checkpoint. It's just a type
+// alias for a string to help differentiate any old string from where a specific state key from the
+// runtime should be used.
+type StateKey string
```

Comment:
I originally had this as a temporary measure to help with making sure I didn't use the wrong string anywhere. It seemed useful enough to leave here after going through that on some of the initial conversions though.